### PR TITLE
Integrates settings api to content/index & improves form validation in social-links form

### LINKED ISSUE #539: Implement Settings API to content/social-links
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

screenshot from 2017-07-22 13-09-28

Work to be done:
- Add Settings API to content/social-links
- Improve validation rules of the social-link-form

### PR DESCRIPTION
**Pull request by geekyd**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
I have split it into two commits as, im trying to achieve two different tasks here. 

#### Changes proposed in this pull request:

- Integrates settings api to content/index
- improves form validation in social-links form

Fixes #539 

Deployment link:
https://open-event-frontend-app.herokuapp.com

### CODE REVIEW

File: app/templates/components/forms/admin/content/social-links-form.hbs [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
   </div>
   <div class="field">
     <label>{{t 'Twitter'}}</label>
-    {{input type='url' name='twitter'}}
+    {{input type='url' name='twitter' value=socials.twitterUrl}}
   </div>
   <div class="field">
     <label>{{t 'Facebook'}}</label>
-    {{input type='url' name='facebook'}}
+    {{input type='url' name='facebook' value=socials.facebookUrl}}
```
Comment (niranjan94) [inline app/templates/components/forms/admin/content/social-links-form.hbs]: Use the url input widget (component)

Comment (geekyd) [inline app/templates/components/forms/admin/content/social-links-form.hbs]: @niranjan94 We will need to modify the component. Please consider the following case

screenshot from 2017-07-23 18-52-41

Also the component produces a segmentedLink which is an object containing protocol & the link. 

Please guide, Thanks.

Comment (niranjan94) [inline app/templates/components/forms/admin/content/social-links-form.hbs]: @geekyd which is why we can do this in the model. 

https://github.com/fossasia/open-event-frontend/blob/development/app/models/event.js#L124

Comment (niranjan94) [inline app/templates/components/forms/admin/content/social-links-form.hbs]: And pass that property to the component instead of the link.

**Comment by geekyd** [comment]
@niranjan94 Please review. Thanks

**Comment by geekyd** [comment]
@harshitagupta30 Thanks for pointing that out. There was a typo while passing the ID to the componenets. I have updated the deployment. Please review. Thanks!

**Comment by sumedh123** [comment]
@geekyd The issue is still there I guess. Please check if you have updated the deployment.

**Comment by geekyd** [comment]
@sumedh123 The deployment link is updated. Please check again. Thanks

**Comment by geekyd** [comment]
@niranjan94 Please review. Thanks

**Pull request merged**